src/components/ShareSaveQuery/shortenApi.js

```diff
@@ -2,31 +2,38 @@ import urlJoin from 'url-join';
 import sqonToName from 'common/sqonToName';
 import { shortUrlApi } from 'common/injectGlobals';
 
-export default ({ stats, queryName, sqon, loggedInUser }) => {
+export default ({ stats, queryName, sqon, loggedInUser, api }) => {
   let { Files, Participants, Families, Size } = stats;
   let alias = queryName || sqonToName({ filters: sqon });
 
   const jwt = localStorage.getItem('EGO_JWT');
-  // TODO: use ajax service?
-  return fetch(urlJoin(shortUrlApi, 'shorten'), {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json',
-      Authorization: `Bearer ${jwt}`,
+
+  const body = JSON.stringify({
+    userid: (loggedInUser || {}).egoId || 'anonymous',
+    alias,
+    content: {
+      ...stats,
+      longUrl: window.location.href,
+      'og:title': `Kids First: ${alias}`,
+      'og:description': `${Files} Files, ${Participants} Participants, ${Families} Families, ${Size} Size`,
+      'og:image':
+        'https://d3b.center/wp-content/uploads/2018/01/Kids-First-Hero-image-01-2-2000x500.png',
+      'twitter:label1': 'Test Label',
+      'twitter:data1': 'test data',
     },
-    body: JSON.stringify({
-      userid: (loggedInUser || {}).egoId || 'anonymous',
-      alias,
-      content: {
-        ...stats,
-        longUrl: window.location.href,
-        'og:title': `Kids First: ${alias}`,
-        'og:description': `${Files} Files, ${Participants} Participants, ${Families} Families, ${Size} Size`,
-        'og:image':
-          'https://d3b.center/wp-content/uploads/2018/01/Kids-First-Hero-image-01-2-2000x500.png',
-        'twitter:label1': 'Test Label',
-        'twitter:data1': 'test data',
-      },
-    }),
-  }).then(r => r.json());
+  });
+
+  return api
+    ? api({
+        url: urlJoin(shortUrlApi, 'shorten'),
+        body,
+      })
+    : fetch(urlJoin(shortUrlApi, 'shorten'), {
```

Comment:
I don't *think* we need this default now that api is being passed through, I'd be in favour of using `api` without the alternative